webapp/channels/src/components/post_view/gm_conversion_message/gm_conversion_message.tsx

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect} from 'react';
+import {Post} from '@mattermost/types/posts';
+import {renderUsername} from 'components/post_markdown/system_message_helpers';
+import {FormattedMessage, useIntl} from 'react-intl';
+import {useDispatch, useSelector} from 'react-redux';
+import {getMissingProfilesByIds} from 'mattermost-redux/actions/users';
+import {makeGetProfilesByIdsAndUsernames} from 'mattermost-redux/selectors/entities/users';
+import {GlobalState} from 'types/store';
+
+export type Props = {
+    post: Post;
+}
+function GMConversionMessage(props: Props): JSX.Element {
+    const convertedByUserId = props.post.props.convertedByUserId;
+    const gmMembersDuringConversionIDs = props.post.props.gmMembersDuringConversionIDs as string[];
+
+    const dispatch = useDispatch();
+    const intl = useIntl();
+
+    useEffect(() => {
+        dispatch(getMissingProfilesByIds(gmMembersDuringConversionIDs));
+    }, [props.post]);
+
+    const userProfiles = useSelector(
+        (state: GlobalState) => makeGetProfilesByIdsAndUsernames()(
```

Comment:
The result of `makeGetProfilesByIdsAndUsernames` needs to be memoized itself for this to work correctly. I think we've done this in other places by writing it like
```js
const getProfilesByIdsAndUsernames = useRef(makeGetProfilesByIdsAndUsernames());
const userProfiles = useSelector((state) => getProfilesByIdsAndUsernames.current(state, {...}));
```

It'd probably also be a good idea to make a reusable hook to do that, but that's obviously not something we need here

Reply:
Done. I didn't know the make... selector pattern fully. Thanks!